### PR TITLE
Fix false-positive migrations in `addEventListener` and JavaScript variable names

### PR DESCRIPTION
**Pull request by RobinMalfait**
This PR fixes 2 false-positives when running the upgrade tool on a Tailwind CSS v3 project converting it to a Tailwind CSS v4 project.

The issue occurs around migrations with short simple names that have a meaning outside if Tailwind CSS, e.g. `blur` and `outline`.

This PR fixes 2 such cases:

1. The `addEventListener` case:

   ```js
   document.addEventListener('blur', handleBlur)
   ```

   We do this by special casing the `addEventListener(` case and making sure the first argument to `addEventListener` is never migrated.

2. A JavaScript variable with default value:

   ```js
   function foo({ foo = "bar", outline = true, baz = "qux" }) {
     // ...
   }
   ```

   The bug is relatively subtle here, but it has actually nothing to do with `outline` itself, but rather the fact that some quote character came before and after it on the same line...

   One of our heuristics for determining if a migration on these small words is safe, is to ensure that the candidate is inside of a string. Since we didn't do any kind of quote balancing, we would consider the `outline` to be inside of a string, even though it is not.

   So to actually solve this, we do some form of quote balancing to ensure that it's _not_ inside of a string in this case.

Additionally, this PR also introduces a small refactor to the `is-safe-migration.test.ts` file where we now use a `test.each` to ensure that failing tests in the middle don't prevent the rest of the tests from running.

### Test plan

1. Added dedicated tests for the cases mentioned in the issue (#18675).
2. Added a few more tests with various forms of whitespace.

Fixes: #18675

### CODE REVIEW

File: packages/@tailwindcss-upgrade/src/codemods/template/is-safe-migration.ts [hunk 1]
```diff
@@ -213,3 +212,38 @@ const styleBlockRanges = new DefaultMap((source: string) => {
     ranges.push(startTag, endTag)
   }
 })
+
+const BACKSLASH = 0x5c
+const DOUBLE_QUOTE = 0x22
+const SINGLE_QUOTE = 0x27
+const BACKTICK = 0x60
```
Comment (thecrypticace) [inline packages/@tailwindcss-upgrade/src/codemods/template/is-safe-migration.ts:219]: Should we consider that in JS backtick strings can span lines — or just not worry about that unless it comes up?

Comment (RobinMalfait) [reply packages/@tailwindcss-upgrade/src/codemods/template/is-safe-migration.ts:219]: We haven't in the past and nobody had issues with this yet (afaik at least) so I don't think we need to add the additional complexity of this (yet) 🤔

**Review by thecrypticace — APPROVED** [review]
